pkg/storage/sds/edvformatprovider/edvformatprovider.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package edvformatprovider
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+	"github.com/hyperledger/aries-framework-go/pkg/storage/sds"
+	"github.com/hyperledger/aries-framework-go/pkg/storage/sds/edvdocumentprocessor"
+)
+
+const (
+	failOpenUnderlyingStore        = "failed to open underlying store in EDVFormatProvider: %w"
+	failCloseUnderlyingStore       = "failed to close underlying store in EDVFormatProvider: %w"
+	failCloseAllUnderlyingStores   = "failed to close all underlying stores in EDVFormatProvider: %w"
+	failEncryptStructuredDocument  = "failed to encrypt structured document into a encrypted document: %w"
+	failMarshalEncryptedDocument   = "failed to marshal encrypted document into bytes: %w"
+	failPutInUnderlyingStore       = "failed to put encrypted document in underlying store in edvFormatStore: %w"
+	failGetFromUnderlyingStore     = "failed to get encrypted document bytes from underlying store in edvFormatStore: %w"
+	failUnmarshalEncryptedDocument = "failed to unmarshal encrypted document bytes into encrypted document struct: %w"
+	failDecryptEncryptedDocument   = "failed to decrypt encrypted document into a structured document: %w"
+	failDeleteInUnderlyingStore    = "failed to delete key-value pair in underlying store in edvFormatStore: %w"
+	failQueryUnderlyingStore       = "failed to query underlying store in edvFormatStore: %w"
+	payloadKey                     = "payload"
+)
+
+var (
+	errPayloadKeyMissing = errors.New(`the structured document content did not contain the ` +
+		`expected "payload" key`)
+	errPayloadNotAssertableAsByteArray = errors.New("unable to assert the payload value as a []byte")
+)
+
+type marshalFunc func(interface{}) ([]byte, error)
+
+// EDVFormatProvider is an encrypted storage provider that uses EDV document models
+// as defined in https://identity.foundation/secure-data-store/#data-model.
+type EDVFormatProvider struct {
+	underlyingProvider   storage.Provider
```

Comment:
storageProvider or just store instead of underlyingProvider?

Reply:
Renamed it to `storeProvider`.